packages/picasso.js/src/web/text-manipulation/text-metrics.js

```diff
@@ -17,21 +17,31 @@ function setContext() {
   context = context || document.createElement('canvas').getContext('2d');
 }
 
-function setFont(fontSize, fontFamily) {
-  if (contextCache.fontSize === fontSize && contextCache.fontFamily === fontFamily) {
+function isEqual(s1, s2) {
+  return (!s1 && !s2) || s1 === s2;
+}
+
+function setFont({ fontStyle, fontWeight, fontSize, fontFamily }) {
+  if (
+    isEqual(contextCache.fontStyle, fontStyle) &&
+    isEqual(contextCache.fontWeight, fontWeight) &&
+    isEqual(contextCache.fontSize, fontSize) &&
+    isEqual(contextCache.fontFamily, fontFamily)
+  ) {
     return;
   }
-
-  context.font = fontSize + ' ' + fontFamily; // eslint-disable-line
+  context.font = [fontStyle, fontWeight, fontSize, fontFamily].filter((value) => !!value).join(' '); // eslint-disable-line
```

Comment:
Is there any scenario where the output of this does not match the supported string? see https://developer.mozilla.org/en-US/docs/Web/CSS/font

Reply:
As I understand context.font has some kind of validation, so if we set something invalid, it will ignore that thing and it also does reorder a few things like putting font style before font weight (e.g. if we put font weight first and then font style in the array, the result still shows the font style first). Not sure if this helps to answer your question.

Reply:
That is more that I can deduce from the documentation at MDN

Reply:
The formal syntax definition actually seem to suggest that order matters and `font-size` and `font-family` are required. So I would say that this implementation is fine given that we do expect those two attributes to be there but `font-style` and `font-weight` are optional